### DISCUSSION TITLE
docker-compose up -d seqr fail due to unhealth container(elasticsearch)

### DISCUSSION THREAD

**Discussion by cluhaowie** [broadinstitute/seqr#2090] [Q&A]
I'm trying to deploy seqr under the instruction of Local installs using docker-compose. Everything works fine but can't start seqr. Below show the error message:
'ERROR: for seqr  Container "8c9a8b7dee7f" is unhealthy.'

After checking the container list, it seems that only seqr_redis_1 and seqr_postgres_1 are successfully initiated. I'm suspecting the one fail should be the "elasticsearch". I tried different container tag of it but without luck. Do anyone have thoughts on it?

Same error in multiple systems:
Ubuntu 18.04.5 LTS
Debian GNU/Linux 10 (buster)
Red Hat Enterprise Linux Server release 7.9 (Maipo)

**Comment by hanars** [comment]
can you please attach your docker logs for the elasticsearch and seqr containers? 
```
docker-compose logs elasticsearch
docker-compose logs seqr
```

**Reply by mriduljohari** [reply]
Hi,
Was this issue resolved? I am having the same "ERROR: for seqr  Container "6a511f65a61d" is unhealthy."
I checked using docker ps -a and this is elsasticsearch (docker.elastic.co/elasticsearch/elasticsearch:7.16.3)
attached is the log for elasticsearch and seqr
[elasticsearch.txt](https://github.com/broadinstitute/seqr/files/8553430/elasticsearch.txt)
[seqr.txt](https://github.com/broadinstitute/seqr/files/8553440/seqr.txt)

**Reply by hanars** [reply]
It looks like the seqr logs you attached actually got truncated - all I see in the file is "Attaching to ". Can you provide an updated log file? Also, can you provide a copy of the docker compose file you are using, and the full output in the terminal when you try to start the seqr container?

**Reply by mriduljohari** [reply]
Attached now are the correct log files and this is the output
$docker-compose up -d seqr
Starting seqr_postgres_1 ... 
Starting seqr_postgres_1 ... done
Starting elasticsearch   ... done

ERROR: for seqr  Container "68307b043675" is unhealthy.
ERROR: Encountered errors while bringing up the project.
[docker-compose.txt](https://github.com/broadinstitute/seqr/files/8564700/docker-compose.txt)
[elasticsearch_log.txt](https://github.com/broadinstitute/seqr/files/8564701/elasticsearch_log.txt)
[seqr_log.txt](https://github.com/broadinstitute/seqr/files/8564702/seqr_log.txt)

**Reply by hanars** [reply]
So one issue is I am seeing an `AccessDeniedException` in elasticsearch to access the mounted directory for the data. Can you double check what permissions you have set on the `./data/elasticsearch` mounted directory and make sure the docker image will have full access? I'm also seeing some issues when initializing the postgres database. Have you previously installed seqr before, or is this your first time? Either way, can you double check the permissions and space available to the `./data/postgres` mounted directory?

**Reply by mriduljohari** [reply]
Thanks, I was able to get elasticsearch 'healthy' status finally doing the following things:
sudo chown -R 1000:1000 ./data/elasticsearch
sudo chown -R 1000:1000 ./data/postgres

I had to also open the port 9200 for elasticsearch

**Reply by hanars** [reply]
Glad to hear that fixed it
